pkg/server/server_test.go

```diff
@@ -79,6 +87,51 @@ var _ = Describe("ApplicationsServer", func() {
 
 		Expect(res.Application.Name).To(Equal(name))
 	})
+	It("Authorize", func() {
+		ctx := context.Background()
+		provider := "github"
+		token := "token"
+
+		jwtClient := auth.NewJwtClient()
+		expectedToken, err := jwtClient.GenerateJWT(auth.SecretKey, auth.ExpirationTime, gitproviders.GitProviderGitHub, token)
+		Expect(err).NotTo(HaveOccurred())
+
+		res, err := appsClient.Authenticate(ctx, &pb.AuthenticateRequest{
+			ProviderName: provider,
+			AccessToken:  token,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(res.Token).To(Equal(expectedToken))
+	})
+	It("Authorize fails on wrong provider", func() {
+		ctx := context.Background()
+		provider := "wrong_provider"
+		token := "token"
+
+		_, err := appsClient.Authenticate(ctx, &pb.AuthenticateRequest{
+			ProviderName: provider,
+			AccessToken:  token,
+		})
+
+		Expect(err.Error()).To(ContainSubstring(BadErrProvider.Error()))
+		Expect(err.Error()).To(ContainSubstring(codes.InvalidArgument.String()))
+
+	})
+
+	It("Authorize fails on empty provider token", func() {
+		ctx := context.Background()
+		provider := "github"
+
+		_, err := appsClient.Authenticate(ctx, &pb.AuthenticateRequest{
+			ProviderName: provider,
+			AccessToken:  "",
+		})
+
+		Expect(err.Error()).To(ContainSubstring(ErrEmptyAccessToken.Error()))
```

Comment:
Bonus points: I think I'd write a custom matcher for these https://onsi.github.io/gomega/#adding-your-own-matchers that would look like `Expect(err).To(MatchGRPCError(codes.InvalidArgument, ErrEmptyAccessToken))`